File: fizzbuzz/fizzbuzz.rs

fn main() {
  
  println!("Enter any number:");
  // Update later
  fizzer(50);
}

fn fizzer(v: i32) {
  for i in 1..v+1 {
    match (i%3, i%5) {
      (0, 0) => println!("FizzBuzz"),
      (0, _) => println!("Fizz"),
      (_, 0) => println!("Buzz"),
      (_, _) => println!("{}", i)
    }
  }
}

File: fizzbuzz/fizzbuzz.js
const fizzer = (i) => {
  const fizzData = {
    3: "Fizz",
    5: "Buzz",
  }; // This will allow us to only have to edit a single line to update fizzbuzz
  let output = "";

  for (let v in fizzData) {
    if (i % v === 0) output += fizzData[v];
  }

  if (output === "") output = i;
  return output;
};

const main = (v) => {
  for (let i = 1; i <= v; i++) {
    // I really hate when for loops start at 1...
    console.log(fizzer(i));
  }
};

/* To fix in the future
console.log("Welcome to fizzbuzz, please enter a length: ");
var num = readline();
*/

main(10);
